File: c++/NeetCode150/11_two_sum.cpp
#include <unordered_map>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) 
    {
        unordered_map<int, int> numMap;
        for(int i = 0; i < numbers.size(); i++)
        {
            int diff = target - numbers[i];
            if (numMap.find(diff) != numMap.end())
            {
                vector<int> result = {numMap[diff]+ 1, i + 1};
                return result;
            }
            numMap[numbers[i]] = i;
        }
    }
};

File: java/NeetCode150/11_two_sum.java
class Solution {
    public int[] twoSum(int[] numbers, int target) 
    {
        HashMap<Integer, Integer> numMap = new HashMap<Integer, Integer> ();
        for(int i = 0; i < numbers.length; i++)
        {
            Integer diff = target - numbers[i];
            if(numMap.containsKey(diff))
            {
                int[] result = new int[2];
                result[0] = numMap.get(diff) + 1;
                result[1] = i + 1;
                return result;
            }
            numMap.put(numbers[i], i);
        }
         return new int[0];
    }
}
